src/libcollections/dlist.rs

```diff
@@ -546,6 +588,58 @@ impl<T> DList<T> {
     pub fn pop_back(&mut self) -> Option<T> {
         self.pop_back_node().map(|box Node{value, ..}| value)
     }
+
+    /// Splits the list into two at the given index. Returns everything after the given index,
+    /// including the index.
+    ///
+    /// This operation should compute in O(N) time.
+    #[stable]
+    pub fn split_off(&mut self, at: uint) -> DList<T> {
+        let len = self.len();
+        if len <= at {
+            panic!("Cannot split off at a nonexistent index");
+        }
+        if at == 0 {
+            let mut new_list = DList::new();
+            new_list.append_mut(self);
+            return new_list;
+        }
+
+        let mut split_node = if at - 1 <= len - (at - 1) {
+            let mut iter = self.iter_mut();
+            // instead of skipping using .skip() (which creates a new struct),
+            // we skip manually so we can access the head field without
+            // depending on implementation details of Skip
+            for _ in range(0, at-1) {
+                iter.next();
+            }
+            iter.head
+        }  else {
+            // better off starting from the end
+            let mut iter = self.iter_mut();
+            for _ in range(0, len - (at - 1)) {
+                iter.next_back();
+            }
+            iter.tail
+        };
+
+        let mut splitted_list = DList {
+            list_head: None,
+            list_tail: self.list_tail,
+            length: len - at - 1
+        };
+
+        if let Some(node) = split_node.resolve() {
```

Comment:
You can make it an `expect` if you please; personally I don't think it's a big deal.